File: backend/app/schemas/risk.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RiskAssessmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    deal_id: uuid.UUID
    score: int
    band: str
    flags: list | None
    rules_applied: list | None
    recommendation: str | None
    version: int
    created_at: datetime

File: mobile/src/types/risk.ts
export type RiskBand = 'green' | 'orange' | 'red'

export interface RiskAssessment {
  id: string
  deal_id: string
  score: number
  band: RiskBand
  flags: string[] | null
  rules_applied: string[] | null
  recommendation: string | null
  version: number
  created_at: string
}
